lib/angular/directive.js

```diff
@@ -1,91 +1,96 @@
 'use strict';
+var CONSTANTS = require('./constants');
 
-var ngModule = angular.module('wfm.map.directives', ['wfm.core.mediator']);
-module.exports = 'wfm.map.directives';
 
-require('../../dist');
+module.exports = function(config) {
+  config = config || {};
 
-ngModule.directive('workorderMap', function($templateCache, mediator, $window, $document, $timeout) {
-  function initMap(element, center) {
-    var myOptions = {
-      zoom:14,
-      center:new google.maps.LatLng(center[0], center[1]),
-      mapTypeId: google.maps.MapTypeId.ROADMAP
-    };
-    var map = new google.maps.Map(element[0].querySelector('#gmap_canvas'), myOptions);
-    return map;
-  }
+  config.adminMode = config.mode === CONSTANTS.MODES.ADMIN;
 
-  function resizeMap(element, parent) {
-    var mapElement = element[0].querySelector('#gmap_canvas');
-    var height = parent.clientHeight;
-    var width = parent.clientWidth;
-    mapElement.style.height = height + 'px';
-    mapElement.style.width = width + 'px';
+  angular.module(CONSTANTS.MAP_DIRECTIVE, ['wfm.core.mediator'])
+    .directive('workorderMap', function(mediator, $window, $document, $timeout) {
+      function initMap(element, center) {
+        var myOptions = {
+          zoom: config.zoom || 14,
+          center:new google.maps.LatLng(center[0], center[1]),
+          mapTypeId: google.maps.MapTypeId.ROADMAP
+        };
+        var map = new google.maps.Map(element[0].querySelector('#gmap_canvas'), myOptions);
```

Comment:
We might need to take a look at the Google Map APIs for API keys etc. 

Probably outside the scope of this refactor though.